src/platform/nrfconnect/wifi/NrfWiFiDriver.cpp

```diff
@@ -206,77 +213,94 @@ void NrfWiFiDriver::ConnectNetwork(ByteSpan networkId, ConnectCallback * callbac
     WiFiManager::Instance().Connect(mStagingNetwork.GetSsidSpan(), mStagingNetwork.GetPassSpan(), handling);
 
 exit:
-    if (status != Status::kSuccess)
+    if (Instance().mpConnectCallback)
     {
-        mpConnectCallback = nullptr;
-        callback->OnResult(status, CharSpan(), 0);
+        Instance().mpConnectCallback->OnResult(status, CharSpan(), 0);
+        Instance().mpConnectCallback = nullptr;
     }
 }
 
-CHIP_ERROR GetConfiguredNetwork(Network & network)
+void NrfWiFiDriver::LoadFromStorage()
 {
-    return CHIP_NO_ERROR;
+    WiFiNetwork network;
+
+    mStagingNetwork = {};
+    ReturnOnFailure(KeyValueStoreMgr().Get(kSsidKey, network.ssid, sizeof(network.ssid), &network.ssidLen));
+    ReturnOnFailure(KeyValueStoreMgr().Get(kPassKey, network.pass, sizeof(network.pass), &network.passLen));
+    mStagingNetwork = network;
 }
 
-void NrfWiFiDriver::OnConnectWiFiNetwork()
+void NrfWiFiDriver::OnScanWiFiNetworkDone(WiFiManager::WiFiRequestStatus status)
```

Comment:
Is it called in the context of CHIP thread? `OnScanWiFiNetworkResult` suggests it's not, in which case `ScheduleLambda` may be needed.

Reply:
`OnScanWiFiNetworkResult` is registered (with the use of lambda) as a net_mgmt event callback in `WiFiManager::Scan()`
And with my recent commit all net_mgmt event callbacks are handled in CHIP context (`WifiMgmtEventHandler` -> `WifiMgmtEventHandlerImpl`). But please check again, I may be missing something.

Reply:
Ok, then `OnScanWiFiNetworkResult` probably shouldn't use `StackLock`, right?

Reply:
Yes, sorry, I have overlooked that.